src/libs/Notification/PushNotification/index.native.js

```diff
@@ -57,6 +81,11 @@ function pushNotificationEventCallback(eventType, notification) {
 function init() {
     // Setup event listeners
     UrbanAirship.addListener(EventType.PushReceived, (notification) => {
+        // By default, refresh notification opt-in status to true if we receive a notification
+        if (!isUserOptedInToPushNotifications) {
+            PushNotification.setPushNotificationOptInStatus(true);
+        }
```

Comment:
This is a fallback, correct? I

If so, any bug in our web logic would reenable notification, right? That's probably the preferable way to fallback, but I just wanted to check.

Reply:
Yeah, I mean it's really just there as a sanity check. If we receive a push notification, then we know that we must be opted-in to push notifications.